solc/CommandLineInterface.cpp

```diff
@@ -438,16 +438,16 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 					continue;
 				}
 
-				m_sourceCodes[infile.string()] = dev::readFileAsString(infile.string());
-				path = boost::filesystem::canonical(infile).string();
+				m_sourceCodes[infile.generic_string()] = dev::readFileAsString(infile.generic_string());
```

Comment:
Is the second `generic_string` needed here?

Reply:
Or to be more specific - shouldn't it rather use the native format?

Reply:
The first change of `string()` to `generic_string()` is the actual fix to the root cause. I'd recommend you reverting the second call to `generic_string()` again, as this is not changing any meaning.

At a later stage (post-v0.5.0), I'd suggest we change the type of the container from `map<string, string>` to `map<path, string>` to better reflect the meaning of the key and avoid unnecessary conversions (or accidentally forgetting them).